rama-net/src/client/pool/mod.rs

```diff
@@ -125,6 +126,26 @@ impl<C, ID> LeasedConnection<C, ID> {
     }
 }
 
+impl<C: ExtensionsRef, ID> ExtensionsRef for LeasedConnection<C, ID> {
+    fn extensions(&self) -> &Extensions {
+        self.pooled_conn
+            .as_ref()
+            .expect("only None after drop")
+            .conn
+            .extensions()
+    }
+}
+
+impl<C: ExtensionsMut, ID> ExtensionsMut for LeasedConnection<C, ID> {
+    fn extensions_mut(&mut self) -> &mut Extensions {
+        self.pooled_conn
+            .as_mut()
+            .expect("only None after drop")
```

Comment:
FYI, for your backlog, not for this PR, so just read/ack and close as-is, but there are also unsafe primitives that allow you this in a more explicit manner. So we do not need to do expect but instead use unsafe and acknowledge that this is fine. Forget what the primitive is, but there's something in there which does basically the same but which is more clear/explicit than an option and intended for use cases like this